grid2op/Reward/AlertReward.py

```diff
@@ -0,0 +1,263 @@
+                "change the reward class with `grid2op.make(..., reward_class=AnyOtherReward)`"
+            )
+        if not env._has_attention_budget:
+            raise Grid2OpException(
+                'Impossible to use the "AlertReward" with an environment for which this feature '
+                'is disabled. Please make sure "env._has_attention_budget" is set to ``True`` or '
+                "change the reward class with `grid2op.make(..., reward_class=AnyOtherReward)`"
+            )
+        if env.parameters.ASSISTANT_WARNING_TYPE != "BY_LINE":
+            raise Grid2OpException(
+                'Impossible to use the "AlertReward" with an environment for which this feature '
+                'is ``ZONAL``. Please make sure "env.parameters.ASSISTANT_WARNING_TYPE" is set to ``BY_LINE`` or '
+                "change the reward class with `grid2op.make(..., reward_class=AnyOtherReward)`"
+            )
+        self.reset(env)
+
+        if isinstance(env._opponent_class, GeometricOpponentMultiArea): 
+            raise Grid2OpException("GeometricOpponentMultiArea is not handled by the alert feature")     
+
+        self.total_time_steps = env.max_episode_duration()
+        self.time_window = env.parameters.ALERT_TIME_WINDOW
+
+        # Storing attacks in the past time window 
+        self._has_attack_in_time_window = np.full(self.time_window, False, dtype=dt_bool)
+        self._fist_attack_step_in_time_window = np.full(env.dim_alerts, -1, dtype=dt_int) # time steps of the first attack
+        self._attacks_in_time_window = np.full((env.dim_alerts, self.time_window), False, dtype=dt_bool)
+        self._has_attack_at_first_time_in_window = np.full(self.time_window, False, dtype=dt_bool)
+        self._is_first_step_of_attack = np.full((env.dim_alerts, self.time_window), False, dtype=dt_bool)
+
+        # Storing alert in the past time window
+        self._line_alerts_in_time_window = np.full((env.dim_alerts, self.time_window+1), False, dtype=dt_bool)
+        self._has_line_alerts_in_time_window = np.full(self.time_window+1, -1, dtype=dt_int)
+        
+        # Current index 
+        self.current_step_first_encountered = dt_int(0)
+                
+    def _step_update(self, legal_alert_action, action, env): 
+        """Update all 
+
+        Args:
+            legal_alert_action (Grid2opAction): RL action, where illegal actions are filtered by budget
+            env (Grid2opEnv): RL environment
+
+        Raises:
+            Grid2OpException: if the opponent is of type GeometricOpponentMultiArea, has it is not handled by the alert feature
+            Grid2OpException: Attacks on substations are not handled by the agent
+        """
+        # Get number of area 
+        if isinstance(env._opponent_class, GeometricOpponentMultiArea): 
+            self.nb_area = len(self._opponent.list_opponents)
+            raise Grid2OpException("GeometricOpponentMultiArea is not handled by the alert feature")
```

Comment:
Same question as above: is this really not supported (and I have to code it) or is this legacy code not removed ?